scripts/Jenkinsfiles/python-unittest

```diff
@@ -0,0 +1,233 @@
+def runPythonTests() {
+    git_branch = xdist_git_branch()
+    git_refspec = xdist_git_refspec()
+    sshagent(credentials: ['jenkins-worker', 'jenkins-worker-pem'], ignoreMissing: true) {
+        checkout changelog: false, poll: false, scm: [$class: 'GitSCM', branches: [[name: git_branch]],
+            doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'CloneOption', honorRefspec: true,
+            noTags: true, shallow: true]], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'jenkins-worker',
+            refspec: git_refspec, url: "git@github.com:philanthropy-u/${REPO_NAME}.git"]]]
+        console_output = sh(returnStdout: true, script: 'bash scripts/all-tests.sh').trim()
+        dir('stdout') {
+            writeFile file: "${TEST_SUITE}-stdout.log", text: console_output
+        }
+        stash includes: 'reports/**/*coverage*', name: "${TEST_SUITE}-reports"
+    }
+}
+
+def pythonTestCleanup() {
+    archiveArtifacts allowEmptyArchive: true, artifacts: 'reports/**/*,test_root/log/**/*.log,**/nosetests.xml,stdout/*.log,*.log'
+    junit '**/nosetests.xml'
+    sh '''source $HOME/edx-venv/bin/activate
+    bash scripts/xdist/terminate_xdist_nodes.sh'''
+}
+
+def xdist_git_branch() {
+    if (env.ghprbActualCommit) {
+        return "${ghprbActualCommit}"
+    } else {
+        return "${BRANCH_NAME}"
+    }
+}
+
+def xdist_git_refspec() {
+    if (env.ghprbActualCommit) {
+        return "+refs/pull/${ghprbPullId}/*:refs/remotes/origin/pr/${ghprbPullId}/*"
+    } else {
+        return "+refs/heads/${BRANCH_NAME}:refs/remotes/origin/${BRANCH_NAME}"
+    }
+}
+
+pipeline {
+    agent { label "jenkins-worker" }
+    options {
+        timestamps()
+        timeout(120)
+    }
+    environment {
+        TOXENV = "test"
```

Comment:
Can we please switch to single qoutes here and in **all** places in this file, whereever possible?

Reply:
@harris-irfan This can be changed and I've, but at a lot of places in sh file double and single quotes have their own meaning and then there are a few sh file which were already written by edX and we have modified them and edX is already using double quotes there. May I request you to just go through the python files for this quotes consistency test and if there is any problem we can fix that.